supporting-blog-content/why-rag-still-matters/why_rag_still_matters.ipynb

```diff
@@ -0,0 +1,1064 @@
+        "\n",
+        " Also we're pushing documents to \"technical-articles\" index."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Creating index\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "collapsed": true,
+        "id": "wh3sIujjUOxM",
+        "outputId": "5a98d167-f7e5-409a-86db-d218424b6796"
+      },
+      "outputs": [],
+      "source": [
+        "if not es_client.indices.exists(index=index_name):\n",
+        "    # Define a simple mapping for text documents\n",
+        "    mappings = {\n",
+        "        \"mappings\": {\n",
+        "            \"properties\": {\n",
+        "                \"text\": {\"type\": \"text\", \"copy_to\": \"semantic_text\"},\n",
+        "                \"meta_description\": {\"type\": \"keyword\", \"copy_to\": \"semantic_text\"},\n",
+        "                \"title\": {\"type\": \"keyword\", \"copy_to\": \"semantic_text\"},\n",
+        "                \"imported_at\": {\"type\": \"date\"},\n",
+        "                \"url\": {\"type\": \"keyword\"},\n",
+        "                \"semantic_text\": {\n",
+        "                    \"type\": \"semantic_text\",\n",
+        "                },\n",
+        "            }\n",
+        "        }\n",
+        "    }\n",
+        "\n",
+        "    es_client.indices.create(index=index_name, body=mappings)\n",
+        "\n",
+        "    print(f\"Created index '{index_name}'\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Populating index"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "Indexing documents using de Bulk API to Elasticsearch"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "file_path = \"dataset.json\"\n",
+        "\n",
+        "actions = []\n",
+        "\n",
+        "with open(file_path, \"r\") as f:\n",
+        "    documents = json.load(f)\n",
+        "    for doc in documents:\n",
+        "        document = {\n",
+        "            \"_index\": index_name,\n",
+        "            \"_source\": {\n",
+        "                \"text\": doc[\"text\"],\n",
+        "                \"url\": doc[\"url\"],\n",
+        "                \"title\": doc[\"title\"],\n",
+        "                \"meta_description\": doc[\"meta_description\"],\n",
+        "                \"imported_at\": datetime.now(),\n",
+        "            },\n",
+        "        }\n",
+        "\n",
+        "        actions.append(document)\n",
+        "\n",
+        "\n",
+        "res = helpers.bulk(es_client, actions)\n",
```

Comment:
It took me 13 minutes to bulk insert all of the docs on a serverless instance. If I'm an outlier, we can ignore this, but if that's a similar experience, can we try out the parallel bulk method?